extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/taskadapter/PodTemplateTaskAdapter.java

```diff
@@ -138,19 +139,22 @@ public Job fromTask(Task task) throws IOException
       podTemplateSelectStrategy = dynamicConfig.getPodTemplateSelectStrategy();
     }
 
-    PodTemplate podTemplate = podTemplateSelectStrategy.getPodTemplateForTask(task, templates);
+    Pair<String, PodTemplate> podTemplatePair = podTemplateSelectStrategy.getPodTemplateForTask(task, templates);
+    assert podTemplatePair.lhs != null && podTemplatePair.rhs != null;
 
     return new JobBuilder()
         .withNewMetadata()
         .withName(new K8sTaskId(task).getK8sJobName())
         .addToLabels(getJobLabels(taskRunnerConfig, task))
         .addToAnnotations(getJobAnnotations(taskRunnerConfig, task))
+        .addToAnnotations(DruidK8sConstants.TASK_JOB_TEMPLATE, podTemplatePair.lhs)
         .endMetadata()
         .withNewSpec()
-        .withTemplate(podTemplate.getTemplate())
+        .withTemplate(podTemplatePair.rhs.getTemplate())
         .editTemplate()
         .editOrNewMetadata()
         .addToAnnotations(getPodTemplateAnnotations(task))
+        .addToAnnotations(DruidK8sConstants.TASK_JOB_TEMPLATE, podTemplatePair.lhs)
```

Comment:
duplicate line?

Reply:
one line is adding it to the job spec and one to the pod template spec (which will become the created pod).

Reply:
Ah, thanks for the clarification, I had missed the `editTemplate()` invocation.